hammer_time/hammer_time.py

```diff
@@ -33,12 +34,37 @@ def perform(client):
         remove_and_wait(client, new_status.iter_new_machines(old_status))
 
 
+def choose_machine(client):
+    """Choose a machine from the client's model at random.
+
+    :param client: The ModelClient to get machines for.
+    :return: a machine-id.
+    :raises: InvalidActionError if there are no machines to choose from.
+    """
+    status = client.get_status()
+    machines = list(m for m, d in status.iter_machines(containers=False))
+    if len(machines) == 0:
+        raise InvalidActionError('No machines to choose from.')
+    return choice(machines)
+
+
+class RebootMachineAction:
+    """Action that reboots a machine."""
+
+    def generate_parameters(client):
+        return {'machine_id': choose_machine(client)}
+
+    def perform(client, machine_id):
+        """Add and remove many containers using the cli."""
+        old_status = client.get_status()
+        client.juju('ssh', (machine_id, 'sudo', 'reboot'), check=False)
```

Comment:
The reboot is not confirmed, but neither was it confirmed in the Matrix version, so this is no worse.  I don't have wait_for_ports handy, because that's in juju-ci-tools/utility.py, not juju-ci-tools/jujupy/utility.py

Maybe a better test would be checking uptime?  After a reboot, we can expect uptime to be less than it previously was.